### PR TITLE
fix: implement the new design of the search area in the mobile view (resolves #24)

### LINKED ISSUE #24: No visible label for search input on small widths
## Describe the bug
The Search input field for the map does not have a visible label at smaller viewport widths (i.e. primarily on mobile devices) and therefore lacks context. This also causes ambiguity with the site-wide search in the main navigation immediately above it.

### To reproduce
1. Go to https://wecount.inclusivedesign.ca/map on a mobile device or in a browser with a narrow viewport
2. Observe that the search input does not have a visible label

### Expected behavior
The search bar should have a visible label

## Screenshots
Screenshot_20210609-001448_cropped

### Desktop
 - OS: Windows 10 Pro v20H2
 - Browser: Chrome 91, Firefox 89
 - Affected version: 0.0.4

### Smartphone / tablet 
 - Device: Pixel 5
 - OS: Android 11
 - Browser: Chrome 91
 - Affected version: 0.0.4

### PR DESCRIPTION
**Pull request by cindyli**

### CODE REVIEW

File: src/html/template.html [hunk 1]
```diff
@@ -5,8 +5,8 @@
 <div class="fl-mapviz-container">
     <!-- The panel containing the search label & the search box -->
     <div class="fl-mapviz-query-wrapper">
-        <label class="fl-mapviz-query-label" for="search-query">Search by postal code or city</label>
-        <svg>
+        <label class="fl-mapviz-query-label" for="fl-search-query">Search by postal code or city</label>
+        <svg role="presentation">
```
Comment (klown) [inline src/html/template.html:9]: @cindyli When I get a chance, I'll load the page into a browser and see how the a11y tree looks.

Followup:  this looks good.  The label is fixed and the magnifying image isn't in the a11y tree.
👍

File: src/js/autocomplete.js [hunk 2]
```diff
@@ -91,5 +91,6 @@ hortis.autocomplete.render = function (that, container, id, widgetOptions) {
     var togo = accessibleAutocomplete(mergedWidgetOptions);
     // TODO: another one for the bestiary of reuse failures
     $("input", container).attr("spellcheck", false);
+    $("input", container).attr("placeholder", "Search by postal code or city");
```
Comment (klown) [inline src/js/autocomplete.js]: Using `@placeholder` as a way of providing a visual label has a11y problems:
* the placeholder text will vanish as soon as the user starts typing in the `<input>`, possibly affecting persons with cognitive issues,
* the placeholder text will not appear as the label for ATs. That might be okay *if* the` <label>` element is still functional even though it isn't visible.

MDN has a critique of `@placeholder`, the section ["Placeholders are not accessible"](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#placeholders_are_not_accessible)

Followup a couple hours later:  First, the use of placeholder looks odd when the `<label>` is visible in the larger view.  Users see both the label and the placeholder, and they are identical.  Secondly, the checker is still complaining that the input "should have a visible label" in the mobile, although there is an accessible label.

If it's okay to shrink the warning text and still display it in the mobile view, why not do the same of the search field label?  That is, show it right above the search field.  Is that difficult to do?

**Review by klown — CHANGES_REQUESTED** [review]

File: src/js/autocomplete.js (hunk 2 above)
Comment (cindyli) [reply src/js/autocomplete.js]: Thanks for the suggestion, @klown. For now, I make the search label accessible by screen readers on the mobile view. Before, this label was hidden on the mobile so it's not surprising the checker complains that the input "should have a visible label" in the mobile.

Displaying the search label above the search field is not technically difficult. I believe the design removes it from the mobile view is to save space for the map. On a small screen such as iphone 5, the map is fairly small with contents around it.

@uttaraghodke, please let us know what you think. Thanks.

**Comment by uttaraghodke** [comment]
This looks good to me.

**Comment by uttaraghodke** [comment]
This looks good to me.

**Comment by cindyli** [comment]
Reopen as this PR was in progress of the review and merge.

**Comment by cindyli** [comment]
@klown, @uttaraghodke, the new design of the search area has been implemented. Ready for another review.

**Comment by klown** [comment]
@cindyli, the label situation is good in the desktop view, but it's missing entirely in the mobile view.  It's not visible and the checker is indicating that there is no label.  See the attached screen shot.  The search field doesn't look like @uttaraghodke's [version](845378).

MobileView

**Comment by cindyli** [comment]
> @cindyli, the label situation is good in the desktop view, but it's missing entirely in the mobile view. It's not visible and the checker is indicating that there is no label. See the attached screen shot. The search field doesn't look like @uttaraghodke's [version](845378).

This might because it was using a cached version. Can you clear the browser cache and try again? Thanks.

**Comment by klown** [comment]
> This might because it was using a cached version. Can you clear the browser cache and try again? Thanks.

Yes, I did that, but to sure, I cleared it again, but I get the same result.

**Comment by klown** [comment]
> > This might because it was using a cached version. Can you clear the browser cache and try again? Thanks.
> 
> Yes, I did that, but to sure, I cleared it again, but I get the same result.

Based on Element conversation with @cindyli, updating the styles by running `npm run build:scss` fixes the problem.

LGTM -- I'll finish the review.

**Review by klown — APPROVED** [review]

**Pull request merged**